controllers/applicationvolumereplication_controller.go

```diff
@@ -195,658 +276,895 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+			if err != nil {
+				return nil, fmt.Errorf("failed to create cloned placementrule error: %w", err)
+			}
+		} else {
+			r.Log.Error(err, "Failed to get avr placementRule", "name", clonedPlRuleName)
 
-			return nil, requeue
+			return nil, fmt.Errorf("failed to get placementrule error: %w", err)
 		}
+	}
+
+	return clonedPlRule, nil
+}
 
-		a.log.Info("Unpausing subscription", "name", subscription.Name)
+func (r *ApplicationVolumeReplicationReconciler) clonePlacementRule(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication, userPlRule *plrv1.PlacementRule,
+	clonedPlRuleName string) (*plrv1.PlacementRule, error) {
+	r.Log.Info("Creating a clone placementRule from", "name", userPlRule.Name)
 
-		err := a.unpauseSubscription(subscription)
-		if err != nil {
-			a.log.Error(err, "failed to unpause subscription", "name", subscription.Name)
+	clonedPlRule := &plrv1.PlacementRule{}
 
-			return nil, requeue
-		}
+	userPlRule.DeepCopyInto(clonedPlRule)
+
+	clonedPlRule.Name = clonedPlRuleName
+	clonedPlRule.ResourceVersion = ""
+	clonedPlRule.Spec.SchedulerName = ""
+
+	err := r.addClusterPeersToPlacementRule(ctx, avr, clonedPlRule)
+	if err != nil {
+		r.Log.Error(err, "Failed to add cluster peers to cloned placementRule", "name", clonedPlRuleName)
 
-		// Subscription has been unpaused. Stop processing it and wait for the next Reconciler iteration
-		a.log.Info("Subscription is unpaused. It will be processed in the next reconciler iteration",
-			"name", subscription.Name)
+		return nil, err
+	}
+
+	err = r.Create(ctx, clonedPlRule)
+	if err != nil {
+		r.Log.Error(err, "failed to clone placement rule", "name", clonedPlRule.Name)
 
-		return nil, requeue
+		return nil, errorswrapper.Wrap(err, "failed to coone PlacementRule")
 	}
 
-	a.log.Info("Subscription is unpaused", "name", subscription.Name)
+	return clonedPlRule, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) addClusterPeersToPlacementRule(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication, plRule *plrv1.PlacementRule) error {
+	clPeersRef := avr.Spec.DRClusterPeersRef
+	clusterPeers := &rmn.DRClusterPeers{}
 
-	exists, err := a.vrgManifestWorkAlreadyExists(subscription)
+	err := r.Client.Get(ctx,
+		types.NamespacedName{Name: clPeersRef.Name}, clusterPeers)
 	if err != nil {
-		return nil, requeue
+		return fmt.Errorf("failed to get cluster peers using %s/%s. Error (%w)",
+			clPeersRef.Name, clPeersRef.Namespace, err)
 	}
 
-	if exists {
-		return nil, !requeue
+	if len(clusterPeers.Spec.ClusterNames) == 0 {
+		return fmt.Errorf("invalid DRClusterPeers configuration. Name %s", clusterPeers.Name)
 	}
-	// This subscription is ready for manifest (VRG) creation
-	placementDecision, err := a.processUnpausedSubscription(subscription)
-	if err != nil {
-		a.log.Error(err, "Failed to process unpaused subscription", "name", subscription.Name)
 
-		return nil, requeue
+	for idx := range clusterPeers.Spec.ClusterNames {
```

Comment:
(future) There is no validation on cluster names in clusterPeers for any mis-configuration. If we choose to not use MCSet (which may be the way forward), we could add some checks here for the same maybe.